src/test/java/tech/picnic/jolo/UtilTest.java

```diff
@@ -10,31 +8,6 @@
 import org.junit.jupiter.api.Test;
 
 public final class UtilTest {
-  /**
-   * In order to test that jOOQ does not perform unexpected mapping of equally named fields in
-   * different tables, we currently use an explicit comparison of the fields' qualified names.
-   * Originally the implementation would just check {@link Field#equals}, but it turns out that this
-   * operator is not symmetric. If this is ever fixed in jOOQ, we should reconsider the
-   * implementation.
-   *
-   * @see <a href="https://github.com/jOOQ/jOOQ/issues/8509">jOOQ bug report</a>
-   */
-  @Test
-  public void testEqualsOnFields() {
-    Field<Long> withSchema = DSL.field(DSL.name("PUBLIC", "FOO", "ID"), Long.class);
-    Field<Long> noSchema = DSL.field(DSL.name("FOO", "ID"), Long.class);
-    assertTrue(FOO.ID.equals(withSchema));
-    assertFalse(withSchema.equals(FOO.ID));
-    assertFalse(FOO.ID.equals(noSchema));
-    assertFalse(noSchema.equals(FOO.ID));
-
-    // Also documenting this oddity: the qualified name of FOO.ID is not "PUBLIC.FOO.ID", even
-    // though it is only considered equal to a field that has the "PUBLIC" qualifier.
-    assertEquals(FOO.ID.toString(), "\"PUBLIC\".\"FOO\".\"ID\"");
-    assertEquals(FOO.ID.getQualifiedName(), DSL.name("FOO", "ID"));
-    assertEquals(withSchema.getQualifiedName(), DSL.name("PUBLIC", "FOO", "ID"));
-    assertEquals(noSchema.getQualifiedName(), DSL.name("FOO", "ID"));
-  }
```

Comment:
This has been fixed upstream. The following tests would now pass.
I have changed the implementation to now use `Field#equals()` directly.

```suggestion
   * @see <a href="https://github.com/jOOQ/jOOQ/issues/8509">jOOQ bug report</a>
   */
  @Test
  public void testEqualsOnFields() {
    Field<Long> withSchema = DSL.field(DSL.name("PUBLIC", "FOO", "ID"), Long.class);
    Field<Long> noSchema = DSL.field(DSL.name("FOO", "ID"), Long.class);
    assertFalse(FOO.ID.equals(withSchema));
    assertFalse(withSchema.equals(FOO.ID));
    assertTrue(FOO.ID.equals(noSchema));
    assertTrue(noSchema.equals(FOO.ID));

    // Also documenting this oddity: the qualified name of FOO.ID is not "PUBLIC.FOO.ID", even
    // though it is only considered equal to a field that has the "PUBLIC" qualifier.
    assertEquals(FOO.ID.toString(), "\"FOO\".\"ID\"");
    assertEquals(FOO.ID.getQualifiedName(), DSL.name("FOO", "ID"));
    assertEquals(withSchema.getQualifiedName(), DSL.name("PUBLIC", "FOO", "ID"));
    assertEquals(noSchema.getQualifiedName(), DSL.name("FOO", "ID"));
  }
```

cc @scranen

Reply:
OK, so the semantics changed but is still a bit strange. `assertFalse(FOO.ID.equals(withSchema))` seems unexpected; can this cause any issues?

Reply:
It's because the [default schema](https://www.jooq.org/doc/3.15/manual/code-generation/codegen-ddl/) assumed by Jooq's code generator is empty, hence `FOO.ID.toString()` (`Field#equals` is based on the String representation) returns `"FOO"."ID"` and the check fails.

If I declare the default schema used by `jooq-code-gen` to be public

```xml
<property>
   <key>unqualifiedSchema</key>
   <value>public</value>
</property>
```

then the following test passes:

```java
    @Test
    public void testEqualsOnFields() {
        Field<Long> withSchema = DSL.field(DSL.name("PUBLIC", "FOO", "ID"), Long.class);
        Field<Long> noSchema = DSL.field(DSL.name("FOO", "ID"), Long.class);

        assertTrue(FOO.ID.equals(withSchema));
        assertTrue(withSchema.equals(FOO.ID));
        assertFalse(FOO.ID.equals(noSchema));
        assertFalse(noSchema.equals(FOO.ID));
        
        assertEquals(FOO.ID.toString(), "\"PUBLIC\".\"FOO\".\"ID\"");
        // this is now unexpected?
        assertEquals(FOO.ID.getQualifiedName(), DSL.name("FOO", "ID"));
        assertEquals(withSchema.getQualifiedName(), DSL.name("PUBLIC", "FOO", "ID"));
        assertEquals(noSchema.getQualifiedName(), DSL.name("FOO", "ID"));
    }
```

I think the H2 in-memory DB does not create a default `public` schema like Postgres does, hence the confusion.

Reply:
To conclude, fields from tables with the same name but from different schemas are considered different.
Given that schemas name-space tables, that makes sense to me.